lib/postgres.go

```diff
@@ -1903,6 +1987,31 @@ func (postgres *Postgres) GetNFTBid(nftPostHash *BlockHash, bidderPKID *PKID, se
 	return &bid
 }
 
+//
+// Derived Keys
+//
+
+func (postgres *Postgres) GetDerivedKey(ownerPublicKey *PublicKey, derivedPublicKey *PublicKey) *PGDerivedKey {
+	key := PGDerivedKey{
+		OwnerPublicKey: *ownerPublicKey,
+		DerivedPublicKey: *derivedPublicKey,
+	}
+	err := postgres.db.Model(&key).WherePK().First()
+	if err != nil {
+		return nil
+	}
+	return &key
+}
+
+func (postgres *Postgres) GetAllDerivedKeysForOwner(ownerPublicKey *PublicKey) []*PGDerivedKey {
+	var keys []*PGDerivedKey
+	err := postgres.db.Model(&keys).Where("owner_public_key = ?", *ownerPublicKey).Select()
+	if err != nil {
+		return nil
+	}
+	return keys
+}
+
```

Comment:
All the Postgres stuff looks fine. Were you able to test it?

Reply:
I think @maebeam knows an environment variable you can set to run the test. Or maybe our continuous integration does it for you.

Reply:
Yeah, tested it manually. It would be ideal to add another unit test to `block_view_test.go` for Postgres, but to do this I'd need a way to generate a test Postgres Db, similarly to `GetTestBadgerDb()` from `blockchain_test.go`. Idk, is there a function like this already?

Reply:
No, but the tests that run via the continuous integration exercise Postgres so no need to do anything beyond a single manual sanity check.